src/hooks/useTableConfig.ts

```diff
@@ -1,23 +1,26 @@
 // src/hooks/useTableConfig.ts
 import { useMemo } from 'react'
 import { ColumnConfig, ColumnConfiguration } from '../helpers/columnConfiguration'
+import { DataFrame } from '../helpers/dataframe'
 
 export interface ColumnDescriptor extends ColumnConfig {
   key: string; // column name
   index: number; // position in current order
 }
 
 export function useTableConfig(
-  header: string[],
+  df: DataFrame,
   config?: ColumnConfiguration
 ): ColumnDescriptor[] {
   return useMemo(() => {
+    const { header, sortable: dfSortable } = df
     const inHeader = new Set(header)
 
     // Build descriptors following DataFrame.header order
     const cols: ColumnDescriptor[] = header.map((key, i) => ({
       key,
       index: i,
+      sortable: dfSortable ?? false, // Set sortable to dataframe's value by default
```

Comment:
it means that we will consider a column as sortable if it's set to true in the column configuration, even if the dataframe is not sortable (`df.sortable === false`).
I think we should only set it to true if `df.sortable === true`.

Also, you might want to take a look at https://github.com/hyparam/hightable/issues/199#issuecomment-2949139889. `orderBy` and the `sortable` properties should match: we must ensure that we don't sort by column A (or columns C,B,A) if the column A is not set as sortable.

We could merge as is, and hope the user will provide the values correctly, but I think we should check and raise an error (or warn and ignore) when we have a conflict between these two states (orderBy / sortable columns)

Reply:
Yes, I feel like the config should only allow to disable sort UI elements for a column